snp_motion_planning/src/plugins/tasks/tcp_speed_limiter.hpp

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <tesseract_kinematics/core/joint_group.h>
+#include <tesseract_kinematics/core/utils.h>
+#include <tesseract_time_parameterization/core/trajectory_container.h>
+#include <tesseract_environment/environment.h>
+
+namespace snp_motion_planning
+{
+class TCPSpeedLimiter
+{
+public:
+  using Ptr = std::shared_ptr<TCPSpeedLimiter>;
+  using ConstPtr = std::shared_ptr<const TCPSpeedLimiter>;
+
+  TCPSpeedLimiter(tesseract_environment::Environment::ConstPtr env) : env_(env)
+  {
+  }
+
+  // Joint names are assumed to be constant across the entire trajectory
+  bool compute(tesseract_planning::TrajectoryContainer& trajectory, std::vector<std::string> joint_names,
+               const double max_speed, const std::string tcp) const
```

Comment:
I think the class definition here is not necessary; let's just make this a function instead:

```suggestion
  void limitTCPSpeed(tesseract_environment::Environment::ConstPtr env, 
    tesseract_planning::TrajectoryContainer& trajectory,
    const std::string& joint_group,
    const std::string& tcp_frame,
    const double max_speed)
```

Note the changes to the signature:
- pass in the environment
- pass in the name of the joint group instead of the joint names
- don't catch exceptions in the function